packages/manager/src/components/CopyableTextField/CopyableTextField.test.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { renderWithTheme } from 'src/utilities/testHelpers';
+import { CopyableTextField } from './CopyableTextField';
+import { downloadFile } from 'src/utilities/downloadFile';
+import { CopyableTextFieldProps } from './CopyableTextField';
+import userEvent from '@testing-library/user-event';
+
+vi.mock('src/utilities/downloadFile', () => ({
+  downloadFile: vi.fn(),
+}));
+
+const mockLabel = 'label';
+const mockValue = 'Text to Copy';
+const defaultProps: CopyableTextFieldProps = {
+  label: mockLabel,
+  value: mockValue,
+};
+
+const setup = (props = {}) => {
+  return renderWithTheme(<CopyableTextField {...defaultProps} {...props} />);
+};
+
+describe('CopyableTextField', () => {
+  it('should render label and CopyableText', () => {
+    const { getByRole } = setup();
+
+    const textbox = getByRole('textbox', { name: mockLabel });
+
+    expect(textbox).toBeVisible();
+    expect(textbox).toHaveAttribute(
+      'value',
+      expect.stringContaining(mockValue)
+    );
+  });
+
+  it('should render the copy icon button and tooltip and allow user to copy the Text', async () => {
+    const { findByRole, getByRole } = setup();
+
+    const copyIcon = getByRole('button', {
+      name: `Copy ${mockValue} to clipboard`,
+    });
+
+    await userEvent.hover(copyIcon);
+    const copyTooltip = await findByRole('tooltip');
+    expect(copyTooltip).toBeInTheDocument();
+    expect(copyTooltip).toHaveTextContent('Copy');
+
+    await userEvent.click(copyIcon);
+    expect(copyIcon.getAttribute('data-qa-tooltip')).toBe('Copied!');
+  });
+
+  it('should render the download icon button and tooltip and allow user to download the Text File', async () => {
+    const { findByRole, getByRole } = setup({ showDownloadIcon: true });
```

Comment:
I understand the reason for the `setup` abstraction but I think using `renderWithTheme` directly is okay too. Using `renderWithTheme` directly is a common pattern throughout the codebase and I personally like verbosity of this approach, but curious what others think

```suggestion
    const { findByRole, getByRole } = renderWithTheme(
      <CopyableTextField {...defaultProps} showDownloadIcon />
    );
```

Reply:
Agree. Let's stick with the established pattern here.